staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -667,8 +669,39 @@ func (c *Cacher) dispatchEvent(event *watchCacheEvent) {
 	c.startDispatching(event)
 
 	// Since add() can block, we explicitly add when cacher is unlocked.
+
+	// Dispatching event in nonblocking way first, which make faster watchers
```

Comment:
Is it possible to continuously delay delivery to slow watcher(s) ?
In some situation, there maybe even starvation.

Reply:
There is a timeout - if we won't be able to send event, we will simply close the watcher.